coremltools/models/utils.py

```diff
@@ -2252,3 +2252,138 @@ def _make_second_chunk_prog(prog: _mil.Program, op_idx: int) -> _mil.Program:
     return prog
 
 
+def change_output_tensor_type(
+    ml_model: "_ct.models.model.MLModel",
+    from_type: int,
+    to_type: int,
+    function_names: _Optional[_List[str]] = None,
+) -> "_ct.models.model.MLModel":
+    """
+    Change output type of the CoreML model tensor outputs. Supported types are FLOAT16, FLOAT32.
+
+    Parameters
+    ----------
+    ml_model: MLModel
+        A Core ML model that needs to change its output type.
+        Note:
+        - the original model is not modified, the model with updated output types is returned as a new instance.
+        - only an mlProgram is supported (not pipelines, not neural networks).
+
+    from_type:
```

Comment:
Do we not think it's possible that a model might have multiple float32 inputs and we only want to convert some of them to float16?

Reply:
I've made the names of input / output variables part of the util function API, so yes that possibility should be covered now, thanks!